packages/dashboard/src/components/robots/robot-mutex-group-app.tsx

```diff
@@ -0,0 +1,186 @@
+            }
+          }
+          if (robot.mutex_groups.requesting) {
+            for (const requesting of robot.mutex_groups.requesting) {
+              if (updatedMutexGroups[requesting]) {
+                updatedMutexGroups[requesting].requestedBy.push(robotIdentifier);
+              } else if (
+                updatedMutexGroups[requesting] &&
+                !updatedMutexGroups[requesting].requestedBy
+              ) {
+                updatedMutexGroups[requesting].requestedBy = [robotIdentifier];
+              } else {
+                updatedMutexGroups[requesting] = {
+                  name: requesting,
+                  lockedBy: undefined,
+                  requestedBy: [robotIdentifier],
+                };
+              }
+            }
+          }
+        }
+      }
+
+      // Filter intermediate mutex groups which are not locked, but just
+      // requested by robots
+      for (const mutexGroupName of Object.keys(updatedMutexGroups)) {
+        if (!updatedMutexGroups[mutexGroupName].lockedBy) {
+          delete updatedMutexGroups[mutexGroupName];
+        }
+      }
+
+      setMutexGroups(updatedMutexGroups);
+    };
+
+    // Initialize table
+    (async () => {
+      await refreshMutexGroupTable();
+    })();
+
+    // Set up regular interval to refresh table
+    const refreshInterval = window.setInterval(
+      refreshMutexGroupTable,
+      RefreshMutexGroupTableInterval,
+    );
+    return () => {
+      clearInterval(refreshInterval);
+    };
+  }, [rmf]);
+
+  const handleUnlockMutexGroup = React.useCallback<React.MouseEventHandler>(async () => {
+    if (!selectedMutexGroup || !selectedMutexGroup.lockedBy) {
+      return;
+    }
+    const fleet = getFleetFromRobotIdentifier(selectedMutexGroup.lockedBy);
+    const robot = getRobotFromRobotIdentifier(selectedMutexGroup.lockedBy);
+    if (!fleet || !robot) {
+      return;
+    }
+
+    try {
+      if (!rmf) {
+        throw new Error('fleets api not available');
+      }
+
+      await rmf.fleetsApi?.unlockMutexGroupFleetsNameUnlockMutexGroupPost(
+        fleet,
+        robot,
+        selectedMutexGroup.name,
+      );
+      appController.showAlert(
+        'success',
+        `Requested to unlock mutex group ${selectedMutexGroup.name} for ${fleet}:${robot}`,
+      );
+    } catch (e) {
+      appController.showAlert(
+        'error',
+        `Failed to unlock mutex group ${selectedMutexGroup.name} for ${fleet}:${robot}: ${
+          (e as Error).message
+        }`,
+      );
+    }
+    setOpenMutexGroupDialog(false);
+  }, [selectedMutexGroup, rmf, appController]);
+
+  return (
+    <TableContainer>
+      <MutexGroupTable
+        mutexGroups={Object.values(mutexGroups)}
+        onMutexGroupClick={(_ev, mutexGroup) => {
+          setOpenMutexGroupDialog(true);
+          setSelectedMutexGroup(mutexGroup);
+        }}
+      />
+      {openMutexGroupDialog && selectedMutexGroup && selectedMutexGroup.lockedBy && (
```

Comment:
This would cause the dialog to be unmounted early before thee animation is finished, remove these conditions to keep the dialog mounted.

Reply:
https://github.com/open-rmf/rmf-web/pull/933/commits/8da155b9f1ae8f1a248eba852ba82be9d0061468